### PR TITLE
Send rumors state when connecting federaction

### PR DESCRIPTION
**Pull request by arnauds5**
- Send rumors state when connecting federaction
- Send rumors state when receiving a connection from federation
- Send greet server when connecting federaction

### CODE REVIEW

File: be1-go/internal/handler/channel/federation/hfederation/federation.go [hunk 1]
```diff
@@ -366,19 +378,28 @@ func (h *Handler) handleChallenge(msg mmessage.Message, channelPath string,
...
+			federationExpect.LaoId)
 	} else {
+		// Add the socket to the list of server sockets
+		h.sockets.Upsert(socket)
+
+		// Send the rumor state directly to avoid delay while syncing
+		err = h.rumors.SendRumorStateTo(socket)
+		if err != nil {
+			return err
+		}
```
Comment (sgueissa) [inline be1-go/internal/handler/channel/federation/hfederation/federation.go:392]: When a server connect to an other server, the receiving server already notify the hub of a new socket and will already upsert the socket and send the rumor state message so you don't need to do it here

**Review by sgueissa — CHANGES_REQUESTED** [review]

File: be1-go/internal/handler/channel/federation/hfederation/federation.go (hunk 1 above)
Comment (sgueissa) [reply be1-go/internal/handler/channel/federation/hfederation/federation.go:392]: as for now, the federation protocol is using the client socket, therefore we need to do that

**Review by sgueissa — APPROVED** [review]

**Comment by sgueissa** [comment]
LGTM

**Review by K1li4nL — APPROVED** [review]
Blind approval

**Pull request merged**